### PR TITLE
feat: add LinkPreview component (COR-3594)

### PR DESCRIPTION
**Pull request by gillyb**
**Fixes or implements VF-XXX**

### Related PRs

- https://github.com/voiceflow/XXXXXXXXX/pull/123

### Checklist

- [ ] Breaking changes have been communicated, including:
    - New required environment variables
    - Renaming of interfaces (API routes, request/response interface, etc)
- [ ] New environment variables have [been deployed](https://www.notion.so/voiceflow/Add-Environment-Variables-be1b0136479f45f1adece7995a7adbfb)
- [ ] Appropriate tests have been written
    - Bug fixes are accompanied by an updated or new test
    - New features are accompanied by a new test

### CODE REVIEW

File: packages/chat/src/components/LinkPreview/styles.css.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  width: '100%',
+  '::after': {
+    content: '',
+    position: 'absolute',
+    height: '100%',
+    width: '100%',
+    top: 0,
+    right: 0,
+    display: 'block',
+    boxShadow: '0px -1px 0px 0px rgba(0, 0, 0, 0.04) inset',
```
Comment (gillyb) [inline packages/chat/src/components/LinkPreview/styles.css.ts:86]: This is the only way I think you can add an 'inset' shadow over an image (as in Mike's design)

File: packages/chat/src/components/LoadingSpinner/LoadingSpinner.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import clsx from 'clsx';
```
Comment (gillyb) [inline packages/chat/src/components/LoadingSpinner/LoadingSpinner.tsx:1]: Copied this from our component library

File: packages/chat/src/components/LoadingSpinner/styles.css.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
+import { keyframes, style, styleVariants } from '@vanilla-extract/css';
```
Comment (gillyb) [inline packages/chat/src/components/LoadingSpinner/styles.css.ts:1]: Copied this from our component library

File: packages/chat/src/components/LinkPreview/index.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  inline?: boolean;
+}
+
+const LinkPreview = forwardRef<HTMLDivElement, PropsWithRef<ILinkPreviewProps>>((props: any, ref) => {
+  const { image, title, link, inline } = props;
+  const linkType = getImageTypeFromUrl(link);
+  const shadow = linkType === 'custom' && !image;
+
+  return (
+    <div ref={ref} style={{ width: '256px' }} className={linkPreviewContainer({ inline })} {...props}>
```
Comment (mikaalnaik) [inline packages/chat/src/components/LinkPreview/index.tsx:28]: why inline styles?

**Review by mikaalnaik — APPROVED** [review]

File: packages/chat/src/components/LinkPreview/index.tsx (hunk 4 above)
Comment (gillyb) [reply packages/chat/src/components/LinkPreview/index.tsx:28]: you're right! my bad, this was temporary to see it as if it was inside a proper Message component.. 
i'll make a note to fix this

**Pull request merged**